src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -1106,7 +1106,6 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                     }
                     else if (isRMW)
                     {
-                        assert((targetReg == op1Reg) || (targetReg != op2Reg));
```

Comment:
as we spoke offline, did you verify if we put this assert back, do we hit them. These asserts were added conservatively with the idea that we understand the scenario under which this cannot be true. I am wondering if we should remove these as we see scenarios that trigger them.

Reply:
Restored this one